File: Ledger/app/src/main/java/keheira/h/ledger/MainActivity.java
package keheira.h.ledger;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends Activity {
    private ListView listView;
    private ArrayList<Payee> log;
    private ListAdapter adapter;
    private DatabaseHandler db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = (ListView)findViewById(R.id.listView);
        db = DatabaseHandler.getInstance(this);
        getInfo();
    }

    private void deletePerson(){
        listView.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
            @Override
            public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
                db.deletePerson(log.get(position));
                log.remove(position);
                adapter.notifyDataSetChanged();
                return true;
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu){
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Take appropriate action for each action item click
        switch (item.getItemId()) {
            case R.id.add:
                addPerson();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void addPerson() {
        Intent i = new Intent(MainActivity.this, Add_Payee.class);
        startActivity(i);
    }
    private void getInfo(){
        log = db.getPeople();
        if (log.size() != 0) {
            //list adapter
            adapter = new ListAdapter(this, log);
            listView.setAdapter(adapter);
            deletePerson();
        }
    }

    @Override
    public void onResume(){
        super.onResume();
        getInfo();
        adapter.notifyDataSetChanged();
    }
}

File: Ledger/app/src/main/java/keheira/h/ledger/MainActivity.kt
package keheira.h.ledger

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.AdapterView
import android.widget.ListView

import java.util.ArrayList

class MainActivity : Activity() {
    private var listView: ListView? = null
    private var log: ArrayList<Payee>? = null
    private var adapter: ListAdapter? = null
    private var db: DatabaseHandler? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        listView = findViewById<View>(R.id.listView) as ListView
        db = DatabaseHandler.getInstance(this)
        getInfo()
    }

    private fun deletePerson() {
        listView!!.onItemLongClickListener = AdapterView.OnItemLongClickListener { parent, view, position, id ->
            db!!.deletePerson(log!![position])
            log!!.removeAt(position)
            adapter!!.notifyDataSetChanged()
            true
        }
    }

    fun addPerson(view: View) {
        val i = Intent(this@MainActivity, AddPayee::class.java)
        startActivity(i)
    }

    private fun getInfo() {
        log = db!!.people
        if (log!!.size != 0) {
            //list adapter
            adapter = ListAdapter(this, log!!)
            listView!!.adapter = adapter
            deletePerson()
        }
    }

    public override fun onResume() {
        super.onResume()
        getInfo()
        if(adapter != null)
            adapter!!.notifyDataSetChanged()
    }
}
